src/components/core-player/CorePlayer.tsx

```diff
@@ -34,6 +34,10 @@ export interface CorePlayerProps {
 	/** `CorePlayer` initial state */
 	initialState?: CorePlayerInitialState;
 	children: ReactNode;
+	/** Time in sec for values that will set `onAlarmUpdate` */
+	timeAlarm?: number[];
+	/** Time before `onAlarmUpdate` event should ran in sec */
```

Comment:
Q: Are I'm right to assume we have this `timeBeforeAlaram` to handle that the caller has delays (i.e. we want to be informed about the next word a bit ahead, so that the app has time to render the decoration in time for when the word really plays).
If so: It might be better to move the functionality into the caller - instead of the media-player. For example, c-app-ui could provide `timeAlarm=[4]` (instead of `timeAlaram=[5]`) if it knows that it takes 1sec to render the decorations. This seems to be easy to do for the caller and it avoids unnecessary features in media-player.

Reply:
@ronnyroeller , 
Yes, a delay (in case of we would like to run it earlier ).
For example, we have alarms:

```
alarms = [0, 1, 2, 3, 4, 5, 6]
timeBeforeAlarm = 1
```

so alarm will be ran at `[0, 1, 2, 3, 4, 5]` seconds 

I've also thinking, maybe these delays to be set in the c-app-ui, but player just to emit without delays?

Reply:
> I've also thinking, maybe these delays to be set in the c-app-ui, but player just to emit without delays?

Yeah, that was my thinking. Instead of having it as a general feature (in video-player) - make it a c-app-ui specific feature there.